src/main/java/mate/academy/secure/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,45 @@
+package mate.academy.secure;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFromDb = userService.getByEmail(email);
+        if (userFromDb.isEmpty()) {
+            throw new AuthenticationException("Can't authenticate user");
+        }
+        User user = userFromDb.get();
+        String hashedPassword = HashUtil.hashPassword(password, user.getSalt());
+        if (user.getPassword().equals(hashedPassword)) {
+            return user;
+        }
+        throw new AuthenticationException("Can't authenticate user");
+    }
```

Comment:
```suggestion
    public User login(String email, String password) throws AuthenticationException {
        Optional<User> userByEmailOptional = userService.findByEmail(email);
        if (userByEmailOptional.isEmpty() || !userByEmailOptional.get().getPassword()
                .equals(HashUtil.hashPassword(password,
                        userByEmailOptional.get().getSalt()))) {
            throw new AuthenticationException("Incorrect login or password");
        }
        return userByEmailOptional.get();
    }
```

Let's optimize our code a little bit)